lib/livebook/apps.ex

```diff
@@ -186,7 +188,12 @@ defmodule Livebook.Apps do
 
       if Livebook.Notebook.AppSettings.valid?(notebook.app_settings) do
         warnings = Enum.map(warnings, &("Import: " <> &1))
-        deploy(notebook, warnings: warnings)
+
+        case Livebook.Hubs.get_offline_hub() do
+          %{id: ^verified_hub_id} -> deploy(notebook, warnings: warnings)
+          %{id: _} -> :ok
```

Comment:
We should log a warning in this case as well, like "Skipping app deployment at #{path}. The notebook is not verified to come from hub #{hub_name}".